### PR TITLE
Faculty app implemented

### PR DESCRIPTION
**Pull request by giusmahnn**
endpoints, views, model, seriailizer created

### CODE REVIEW

File: faculty/models.py [hunk 1]
```diff
@@ -0,0 +1,14 @@
+from django.db import models
+from account.models import Lecturer
+
+# Create your models here.
+
+class Faculty(models.Model):
+    dept_name = models.CharField(max_length=255, blank=True, null=True)
+    info_description = models.TextField(blank=True, null=True)
+    images = models.ImageField(upload_to='faculty_images/', blank=True, null= True,)
+    other_titles = models.CharField(blank=True, null=True, max_length=50)
```
Comment (davidinmichael) [inline faculty/models.py]: What's this for?

File: faculty/models.py [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+    
+    reps_name = models.OneToOneField(
```
Comment (davidinmichael) [inline faculty/models.py]: There can be more than one instance of a Lecturer as a rep. Also, Students can be rep too.

File: faculty/models.py [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+        Lecturer, on_delete=models.CASCADE)
+    reps_degree = models.CharField(max_length=20, null=True, blank=True)
```
Comment (davidinmichael) [inline faculty/models.py]: Remove this. It will be included in the Lecturer or Student relationship.

File: faculty/views.py [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+
+class FacultyView(APIView):
+    premission_classes = [CourseWriteOrRead]
+
+    def get(self, request):
+        faculty = Faculty.objects.all()
+        if faculty:
+            serializer = FacultySerializer(faculty, many=True)
+            return Response(serializer.data, status.HTTP_200_OK)
+        return Response({"message": "No courses available."}, status=status.HTTP_404_NOT_FOUND)
```
Comment (davidinmichael) [inline faculty/views.py]: Review the message.

File: faculty/views.py [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+            serializer = FacultySerializer(faculty)
+            return Response(serializer.data, status.HTTP_200_OK)
+        return Response({"message": "Department not found."}, status=status.HTTP_404_NOT_FOUND)
+
+    def put(self, request, pk):
+        faculty = self.get_object(pk)
+        if faculty:
+            serializer = FacultySerializer(faculty, data=request.data, partial=True)
+            if serializer.is_valid():
+                serializer.save(last_updated_by=request.user)
```
Comment (davidinmichael) [inline faculty/views.py]: I didn't see this last_updated_by field in the serializer.

File: faculty/views.py [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+                return Response(serializer.data, status=status.HTTP_201_CREATED)
+            return Response(serializer.errors, status=status.HTTP_400_BAD_REQUEST)
+        return Response({"message": "Department not found."}, status=status.HTTP_404_NOT_FOUND)
+    
+    def delete(self, request, pk):
+        # TODO: Deleted courses should be added to a recycle bin
```
Comment (davidinmichael) [inline faculty/views.py:56]: ??

File: faculty/views.py (hunk 5 above)
Comment (giusmahnn) [reply faculty/views.py]: It was a oversight

File: faculty/views.py (hunk 6 above)
Comment (giusmahnn) [reply faculty/views.py:56]: I dont understand this

**Pull request merged**